relayer/src/chain/cosmos.rs

```diff
@@ -1469,7 +1475,18 @@ impl ChainEndpoint for CosmosSdkChain {
             )
             .map_err(Error::grpc_transport)?;
 
-        let request = tonic::Request::new(request);
+        // TODO: Request should be a domain type
+        let version_compatible_request = if self.version_specs.ibc_go_pre_v1_2() {
+            // if < v1.2 overwrite with empty `packet_commitment_sequences` field
+            let mut r = request;
+            r.packet_commitment_sequences = vec![];
+            r
+        } else {
+            // if >= v1.2 leave the request unchanged
+            request
+        };
+
+        let request = tonic::Request::new(version_compatible_request);
```

Comment:
After looking more into this, doing more tests, reading on proto, I think we should undo this change. See [unknowns](https://developers.google.com/protocol-buffers/docs/proto3#unknowns).
It turns out unknown fields are just dropped or retained as unknown fields (depending on the proto and parser versions) but the parser should still be able to work with all fields present in the old proto version. I tested without these checks with old gaia versions and all looks good.
(fwiw for ibc-rs/modules and basecoin, prost [drops](https://github.com/tokio-rs/prost/issues/2) unknown fields iiuc)
Would still keep the `versions_specs` boilerplate to deal with incompatibilities not handled by proto, e.g. `chann-init` with >64 len ports`PortId`-s where we need to check the version of the chain potentially receiving the `chann-try`.